htlcswitch/switch.go

```diff
@@ -1098,6 +1131,52 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 		// what the best channel is.
 		destination := destinations[rand.Intn(len(destinations))]
 
+		// Retrieve the incoming link by its ShortChannelID. Note that
+		// the incomingChanID is never set to hop.Source here.
+		s.indexMtx.RLock()
+		incomingLink, err := s.getLinkByShortID(packet.incomingChanID)
+		if err != nil {
+			s.indexMtx.RUnlock()
```

Comment:
can't this just be moved above `err!=nil` check, slightly more readable

Reply:
yes